SqlInstall.ps1

```diff
@@ -0,0 +1,556 @@
+﻿function Install-SqlServer ([string]$AppVolume, [string]$DataVolume, [string]$LogVolume, [string]$TempVolume, [string]$BackupVolume, [int]$Ctp)
+{
+    <#
+    .SYNOPSIS
+
+    This function will help you to quickly install a SQL Server instance. 
+
+    .DESCRIPTION
+
+    This function will help you to quickly install a SQL Server instance. By scanning your system it will apply a number of best practices to your installation
+    The max DOP and max server memory will be set, the sa account will be disabled while the account used to install SQL Server will be added as sysadmin.
+    The number of TempDB files will be set to the number of cores with a maximum of eight.
+    The perform volume maintenance right will be granted to the SQL Server account. If you run this script in an environment where you are not allowed to do this,
+    please revert that operation by removing the right from the local security policy (secpol.msc).
+
+    The autogrowth of file will be set to 64MB, the Cost Threshold for Parallelism will be set to 40. 
+
+    Note that the dowloaded installation file must be unzipped, an ISO has to be mounted. This will not be executed from this script.
+    
+    .PARAMETER Name
+
+    $AppVolume will hold the volume letter of the application disc. If left empty, it will default to C, unless there is a drive named like App
+
+    $DataVolume will hold the volume letter of the Data disc. If left empty, it will default to C, unless there is a drive named like Data
+
+    $LogVolume will hold the volume letter of the Log disc. If left empty, it will default to C, unless there is a drive named like Log
+
+    $TempVolume will hold the volume letter of the Temp disc. If left empty, it will default to C, unless there is a drive named like Temp
+
+    $BackupVolume will hold the volume letter of the Backup disc. If left empty, it will default to C, unless there is a drive named like Backup
+
+    $Ctp will hold the value of your Cost Threshold for Parallelism. If left empty, it will default to 40 (the mean between OLTP(30) and Datawarehousing(50))
+
+    .Inputs
+    None
+
+    .Outputs
+    None
+
+    .Example
+    C:\PS> Install-SqlServer
+
+    This will run the installation with the default settings
+
+    C:\PS> Install-SqlServer -AppVolume "G"
+
+    This will run the installation with default setting apart from the application volume, this will be redirected to the G drive.
+
+
+    #>
+
+    $startScript = @'
```

Comment:
We always try to accommodate as many scenarios as possible. Looks like this one is just for SQL Server 2014. I'd recommend finding the config files for various sqls (SPADE has some), placing them in the bin\installtemplate directory, then doing find/replace on them as needed (`$configini = Get-Content "$script:PSModuleRoot\bin\installtemplate\$version.ini"` then `$configini = $configini.Replace("--placeholder-role--","DatabaseEngine_only")`

Reply:
The script has been tested with SQL2014 and 2017, but that's just one test. 
I agree that using the ini files from each version is a better way to go. 
Do you want support voor 2008/2008R2? Or do we start from 2012 up to 2017 or even 2019?
This because i want to be able to validate the user input and prevent an error on input like '2015'

Reply:
if 2008 causes too much strife, I'd be happy with 2012+. We can use `[ValidateSet("2012", "2014", "2016","2017","2019")]` on that parameter and it'll allow auto-complete 💯

Reply:
ValidateSet is cool!